templates/test/ci/prow-user-assigned-identity/kustomization.yaml

```diff
@@ -0,0 +1,23 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: default
+resources:
+  - ../../../flavors/external-cloud-provider
```

Comment:
you could base this on the prow external cloud provider flavor so you don't have to re-add all the patches below

Reply:
Although that could confuse things when we make changes to the test/ci prow external cloud provider template. You'd have to run `make generate-flavors` twice to be sure.

Reply:
why would you have to run `make generate-flavors` twice? if you make changes to external-cloud-provider-prow base or patch yamls it would automatically get picked up when you run `make generate-flavor` by both templates

Reply:
Another thing to consider: instead of adding a separate template for user-assigned identity, thoughts on making the existing external-cloud-provider prow template/test use user-assined id ?

Reply:
> why would you have to run `make generate-flavors` twice?

I guess I was imagining basing it off the generated external-cloud-provider template and assuming a one-pass generator, but I was probably wrong. I'll take another look; it would be preferable not to grow the list of templates and tests.